### DISCUSSION TITLE
Ebook follow along with the audio book

### DISCUSSION THREAD

**Discussion by JohnnyS** [advplyr/audiobookshelf#2241] [Ideas]
Wondering if this has been discussed or mentioned, I couldnt really find a search for it with what I put but I was wondering if it is possible to implement the ability to have the ebook follow along the audio book as well instead of manually flipping the pages.

**Comment by rabelux** [comment]
That sounds similar to #2091, or are you talking about something different?

**Reply by nichwall** [reply]
This is slightly different, because #2091 is talking about syncing when not using the book (so if you were reading, stopped, and then wanted to listen from the same point, or the other way around).
This one is that the page turns automatically while listening at the same time.

This has been brought up before (may have just been in Discord since I can't find it anywhere in either GH repo), along with a similar request to highlight the text as it's spoken. Both of these will be really hard to implement, since the syncing while not reading doesn't need to be word accurate, but syncing like this while reading needs to be word accurate and everyone's files are not the same (encoded the file, trimmed a second here or there, bitrate, etc). The word level sync would also be required to handle various screen/font sizes (and line spacing).

There are some other projects (like Read beyond) which came up with a modified epub format to support highlighting, but most haven't been maintained for several years.

**Reply by JohnnyS** [reply]
Yeah something different, more of following along with the audio book while also reading the ebook. Yeah highlighting the text while its spoken is the best way to describe it. 

> There are some other projects (like Read beyond) which came up with a modified epub format to support highlighting, but most haven't been maintained for several years.

Aww well it's not too crazy, just wondered if it was possible or not

**Reply by nichwall** [reply]
That's not to say it will *never* be implemented, but it will be a while if it does get implemented.
